### PR TITLE
Validator major update

### PR DESCRIPTION
**Pull request by teshaTe**
VALIDATOR UPDATE: 

- restructuring of the project;
- adding support for ply data format;
- swapping out gaussian diff. rast. pipeiine with gsplat and updating all rendering pipeline accordingly;
- updated instillation, cleaning and updating scripts for validator;
- updated tests
- fixed some bugs in VRAM memory checking;
- updated rendering: now we render half of the views with predefined angles and another half with random angles;

### CODE REVIEW

File: validation/requirements.txt [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-scikit-video==1.1.11
-scikit-learn==1.5.0
-loguru==0.7.2
\ No newline at end of file
+h5py==3.10.*
+numpy==1.26.*
+torch==2.3.*
+transformers==4.41.*
+pytest==8.2.*
+bittensor==6.9.3
```
Comment (the-mx) [inline validation/requirements.txt:6]: Why do we need bittensor? And bring back fastapi and uvicorn

File: validation/serve.py [hunk 2]
```diff
@@ -28,10 +34,10 @@ def get_args():
 
 class RequestData(BaseModel):
     prompt: constr(max_length=1024)
-    data: constr(max_length=100 * 1024 * 1024)
+    data: constr(max_length=1000 * 1024 * 1024)
     # 0 - dream graussian prj data
     # 1 - LGM / other prj data
-    data_ver: int = 0
+    data_ver: int
```
Comment (the-mx) [inline validation/serve.py:40]: default value?

File: validation/serve.py [hunk 3]
```diff
@@ -66,38 +71,99 @@ async def validate(request: RequestData) -> ResponseData:
...
+    - ResponseData: An instance of ResponseData containing the scores generated from the validation process.
+
+    """
+    prompt = request.prompt
+    data = request.data
+    loader = PlyLoader()
+    score = 0.0
+
+    try:
+        score = _validate(prompt, data, 1, loader)
```
Comment (the-mx) [inline validation/serve.py:162]: data_ver = 1?

File: validation/serve.py [hunk 4]
```diff
@@ -28,10 +34,10 @@ def get_args():
 
 class RequestData(BaseModel):
     prompt: constr(max_length=1024)
-    data: constr(max_length=100 * 1024 * 1024)
+    data: constr(max_length=1000 * 1024 * 1024)
```
[the remark this answers is not in the document]
Comment (teshaTe) [reply validation/serve.py:37]: that was set up by Ruben for some reason

File: validation/setup_env.sh [hunk 5]
```diff
@@ -11,32 +11,21 @@ fi
...
-pip install ./extras/diff-gaussian-rasterization
-rm -rf ./extras
+pip install -r requirements.txt
 
 # Store the path of the Conda interpreter
 CONDA_INTERPRETER_PATH=$(which python)
 
 # Generate the validation.config.js file for PM2 with specified configurations
-cat <<EOF > validation.config.js
+cat <<EOF > ../validation.config.js
```
Comment (the-mx) [inline validation/setup_env.sh:28]: remove ..

File: validation/serve.py (hunk 3 above)
Comment (teshaTe) [reply validation/serve.py:162]: yes, according to Ruben's comment

File: validation/serve.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (the-mx) [reply validation/serve.py:37]: we can make it 100 and change it to 10 in the future

File: validation/serve.py (hunk 2 above)
Comment (teshaTe) [reply validation/serve.py]: 0, fixed

File: validation/serve.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (teshaTe) [reply validation/serve.py]: set to 500 now

**Pull request merged**